File: Lista de exercicio/projetoPYTHON/questao7/questao7.py
class Professor:
    def __init__(self, nome):
        self.nome = nome
        self.escolas = []

    def adicionar_escola(self, escola):
        self.escolas.append(escola)

class Escola:
    def __init__(self, nome):
        self.nome = nome
        self.professores = []

    def adicionar_professor(self, professor):
        self.professores.append(professor)
        professor.adicionar_escola(self)

# Exemplo de uso:

def executar_questao7():
    professor1 = Professor("Ricardo")
    professor2 = Professor("Douglas")
    professor3 = Professor("Leonardo")

    escolaA = Escola("escola A")
    escolaB = Escola("escola B")
    
    escolaA.adicionar_professor(professor1)
    escolaB.adicionar_professor(professor2)
    escolaB.adicionar_professor(professor3)
    escolaA.adicionar_professor(professor3)

    print("Professores na Escola A:")
    for prof in escolaA.professores:
        print(prof.nome)

    print("\nProfessores na Escola B:")
    for prof in escolaB.professores:
        print(prof.nome)

File: Lista de exercicio/projetoJAVA/src/projetojava/questao7.java
package projetojava;

import java.util.ArrayList;
import java.util.List;

class Professor {
    String nome;
    List<Escola> escolas;

    Professor(String nome) {
        this.nome = nome;
        this.escolas = new ArrayList<>();
    }

    void adicionarEscola(Escola escola) {
        escolas.add(escola);
    }
}

class Escola {
    String nome;
    List<Professor> professores;

    Escola(String nome) {
        this.nome = nome;
        this.professores = new ArrayList<>();
    }

    void adicionarProfessor(Professor professor) {
        professores.add(professor);
        professor.adicionarEscola(this);
    }
}

public class questao7 {
    public static void main(String[] args) {
        Professor professor1 = new Professor("Ricardo");
        Professor professor2 = new Professor("Douglas");
        Professor professor3 = new Professor("Leonardo");

        Escola escolaA = new Escola("Escola A");
        Escola escolaB = new Escola("Escola B");

        escolaA.adicionarProfessor(professor1);
        escolaB.adicionarProfessor(professor2);
        escolaB.adicionarProfessor(professor3);
        escolaA.adicionarProfessor(professor3);

        System.out.println("Professores na Escola A:");
        for (Professor prof : escolaA.professores) {
            System.out.println(prof.nome);
        }

        System.out.println("\nProfessores na Escola B:");
        for (Professor prof : escolaB.professores) {
            System.out.println(prof.nome);
        }
    }
}
